src/cattr/gen.py

```diff
@@ -181,6 +183,17 @@ def make_dict_structure_fn(cl: Type, converter, **kwargs):
                 f"    res['{ian}'] = {struct_handler_name}(o['{kn}'], __c_t_{an})"
             )
     lines.append("    }")
+    if forbid_extra_fields:
+        post_lines.append("  allowed_fields = {")
+        post_lines += [f"    '{a.name}'," for a in attrs]
+        post_lines += [
+            "  }",
+            "  unknown_fields = set(o.keys()) - allowed_fields",
+            "  if unknown_fields:",
+            "    raise KeyError(",
```

Comment:
I wouldn't use a `KeyError` here. First of all, `KeyError` is raised when a key is missing from a dictionary, not when it's extra :)

Second, `KeyError` is a built-in Python exception. User libraries are supposed either throw `Exception` or something inheriting from `Exception` (see https://docs.python.org/3/library/exceptions.html#Exception). Ideally we would have a `StructureError` and you could inherit from that, but we don't currently and that should be a different task, so just throw `Exception` for now.

Reply:
Hm, I think `KeyError` counts as "something inheriting from `Exception`", but agreed that custom exception subclasses would be best, and that a `KeyError` is a little weird.
```python
try:
    raise KeyError('key error')
except Exception:
    print("caught your key error")
```
I was thinking of it as the key being missing from the *class*, rather than from the dictionary.  But I'm happy to put in `Exception` for now if that's your preference!

Reply:
Yeah, let's switch to a bare `Exception` for now. `KeyError` is a bit of a stretch I think.